### PR TITLE
Add labels documentation to CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by tuckerrc**
This pull request includes some new documentation in the CONTRIBUTING file. Any suggestions for new labels or better ways to organize them are welcome. I'm not sure if this is the best place to discuss new labels but it's a start for what I had in mind. 

(I pulled some of the documentation from the iD repository).

### CODE REVIEW

**Comment by tuckerrc** [comment]
Changes in response to #2927

**Comment by tomhughes** [comment]
As labels can't be set by reporters I'm not sure it's helpful to document them, or at least not here.

**Comment by tomhughes** [comment]
Also I wouldn't just document what we have because that's almost certainly not a good set of labels - it's probably better to tidy them up first and then document them.

The current set probably needs cutting down and we certainly need to decide if multi word labels are going to use space, underscore or hyphen to join words...

**Comment by lectrician1** [comment]
Definitely agree. Could the labels be given descriptions within GitHub itself as well please?

**Comment by tuckerrc** [comment]
Reporters may not be able to add labels but someone looking for an issue to work on may find this documentation helpful. I am okay with the documentation being on the [labels page](https://github.com/openstreetmap/openstreetmap-website/labels) or the CONTRIBUTING.md file. 

A few benefits of putting the documentation here is @lectrician1 and I can make suggestions that only require accepting Pull Requests and anyone looking to contribute will find them when reading CONTRIBUTING.md.

**Comment by tomhughes** [comment]
My main point was questioning whether it should be in the contributor documentation rather than whether it should be documented at all - basically whether there should be separate documentation for maintainers.

**Comment by tuckerrc** [comment]
I think having both can be useful. A simplified one for contributors and a detailed one for maintainers to use when doing triage.

**Comment by gravitystorm** [comment]
@tuckerrc I want to clarify the first tag I've noticed. 'work-in-progress' (and [WIP] in subject lines of pull requests) means that the author knows that their own proposed code is not yet ready to merge (maybe the tests fail, or something else needs to be finished first), but the author of the code is (likely) sharing it in order to gain early feedback. It doesn't mean 'Don't start work on these, somebody else already did!'

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -108,6 +108,65 @@ other commits, namely:
...
+* <sub>[![awaiting upstream][awaiting_upstream]][awaiting_upstream_link]</sub> -
+Awaiting Upstream. These issues won't be resolved until changes are made upstream.
+* <sub>[![enhancement][enhancement]][enhancement_link]</sub> -
+Enhancements. These are features that improve already supported .
+* <sub>[![changes requested][changes_requested]][changes_requested_link]</sub> -
+Changes Requested. There are more changes needed before a pull request will be accepted.
+* <sub>[![dx][dx]][dx_link]</sub> -
+Developer Experience. The things that help with development (Docker, Vagrant, Rake, etc.)
+* <sub>[![work-in-progress][wip]][wip_link]</sub> -
+Work in Progress.  Don't start work on these, somebody else already did!
```
Comment (pnorman) [inline CONTRIBUTING.md:152]: Draft PRs should be used for this.

**Comment by gravitystorm** [comment]
I think this has been superceeded by the work in #3043 where we discussed the current + proposed labels, and also added descriptions in the Github user interface for all of the labels.

I think having the descriptions there are sufficient, and we don't need to duplicate the descriptions in the contributing file - they are likely to get out of sync, and it is also additional reading that new contributors don't really need to worry about so I don't want the whole document to become (more) intimidating.

Thanks for opening the PR though - I think we ended up with an even better result via implementing #3043.

**Comment by tuckerrc** [comment]
> I think we ended up with an even better result via implementing #3043.

I agree. Thanks for adding the new labels!

**Pull request closed without merging**